src/components/scenegraph/Toolbar.js

```diff
@@ -56,8 +56,45 @@ export default class Toolbar extends Component {
 
   componentDidMount() {
     document.addEventListener('click', this.handleClickOutsideSave);
+    this.checkSignInStatus();
   }
 
+  componentDidUpdate(prevProps) {
+    if (this.props.currentUser !== prevProps.currentUser) {
+      this.setState({ currentUser: this.props.currentUser });
+
+      if (
+        localStorage.getItem('pendingSceneSave') === 'true' &&
+        this.props.currentUser
+      ) {
+        localStorage.removeItem('pendingSceneSave');
+        setTimeout(() => this.cloudSaveHandler({ doSaveAs: true }), 500);
```

Comment:
Why settimeout for 500 instead of waiting for a specific event?

Reply:
ah I see to prevent "race condition" where maybe save handler is called twice because it takes time to remove pendingSceneSave from local storage.